src/lib/support/Pool.cpp

```diff
@@ -127,6 +127,46 @@ Loop StaticAllocatorBitmap::ForEachActiveObjectInner(void * context, Lambda lamb
     return Loop::Finish;
 }
 
+size_t StaticAllocatorBitmap::FirstActiveIndex()
+{
+    size_t idx = 0;
+    for (size_t word = 0; word * kBitChunkSize < Capacity(); ++word)
+    {
+        auto & usage = mUsage[word];
+        auto value   = usage.load(std::memory_order_relaxed);
+        for (size_t offset = 0; offset < kBitChunkSize && offset + word * kBitChunkSize < Capacity(); ++offset)
+        {
+            if ((value & (kBit1 << offset)) != 0)
+            {
+                return idx;
+            }
+            idx++;
+        }
+    }
+    VerifyOrDie(idx == mCapacity);
```

Comment:
This looks dangerous, in case iteration is done wrong, you can assert. Probably OK, but it makes me concerned.

Reply:
This will never trigger. At the same time I agree with the comment being "this is not obviously never triggerring..." ... which is exactly why I added it. This piece of nested for was taken from existing code (so I do not mess it up) and then I added the assert because the code is messy.